sqlglot/optimizer/qualify_columns.py

```diff
@@ -42,12 +43,16 @@ class Context:
         aliased_columns: List of column names.
             This should be set for derived tables or CTEs where the parent context
             defines column names for an alias, e.g. "SELECT * FROM (SELECT * FROM x) AS y(a, b)"
+        correlated_selectables: Mapping of name to selectable columns.
```

Comment:
i saw you moved out class Selectable, do we need to move out Context as well for other optimization functions?

Reply:
I moved out "SelectScope".... not super crazy about the name.

In this case, "Context" is specific to this rule.

I feel like there's potentially a bigger, better abstraction here. I haven't quite nabbed it yet, though

Reply:
For SelectScope naming: you can perhaps rename to SelectParts, or consider moving it inside select class as property methods e.g. tables, columns etc